libs/application-generic/src/usecases/upsert-preferences/upsert-preferences.usecase.ts

```diff
@@ -110,7 +110,14 @@ export class UpsertPreferences {
         throw new BadRequestException('Preference not found');
       }
 
-      return this.deletePreferences(command, foundPreference?._id);
+      await this.deletePreferences(command, foundPreference?._id);
+
+      /*
+       * Ideally we need to return the foundPreference with a deleted: true flag
```

Comment:
Adding a TODO flag for easier searching
```suggestion
       * TODO: Ideally we need to return the foundPreference with a deleted: true flag
```